requirements-common.txt

```diff
@@ -20,6 +20,7 @@ outlines >= 0.0.43, < 0.1 # Requires torch >= 2.1.0
 typing_extensions >= 4.10
 filelock >= 3.10.4 # filelock starts to support `mode` argument from 3.10.4
 pyzmq
+msgspec
```

Comment:
I remember I once tried this library, but its serialization scopt is quite limited. Some classes cannot be serialized via this library. Do you have this experience when use it in ray?

Reply:
we used this library for our internal fork before. For this PR, I have to implement custom reduce for `array` type. And Union of the same 2 types are not supported (e.g., Union[OrderedDict, Dict] kind of thing). But I think we can get around this much